### PR TITLE
Implement `debug_more_non_exhaustive`

### PR DESCRIPTION
**Pull request by tgross35**
This implements the ACP at https://github.com/rust-lang/libs-team/issues/248, adding `.finish_non_exhaustive()` for `DebugTuple`, `DebugSet`, `DebugList`, and `DebugMap`.

Also used this as an opportunity to make some documentation and tests more readable by using raw strings instead of escaped quotes.

Tracking issue: https://github.com/rust-lang/rust/issues/127942

### CODE REVIEW

File: library/core/src/fmt/builders.rs [hunk 1]
```diff
@@ -697,6 +792,55 @@ impl<'a, 'b: 'a> DebugList<'a, 'b> {
         self
     }
 
+    /// Marks the struct as non-exhaustive, indicating to the reader that there are some other
+    /// fields that are not shown in the debug representation.
```
Comment (Nilstrieb) [inline library/core/src/fmt/builders.rs:796]:
```suggestion
    /// Marks the list as non-exhaustive, indicating to the reader that there are some other
    /// elements that are not shown in the debug representation.
```

File: library/core/src/fmt/builders.rs [hunk 2]
```diff
@@ -554,6 +599,56 @@ impl<'a, 'b: 'a> DebugSet<'a, 'b> {
...
```
Comment (Nilstrieb) [inline library/core/src/fmt/builders.rs:603]:
```suggestion
    /// Marks the set as non-exhaustive, indicating to the reader that there are some other
    /// elements that are not shown in the debug representation.
```

File: library/core/src/fmt/builders.rs [hunk 3]
```diff
@@ -973,6 +1117,62 @@ impl<'a, 'b: 'a> DebugMap<'a, 'b> {
...
```
Comment (Nilstrieb) [inline library/core/src/fmt/builders.rs:1121]:
```suggestion
    /// Marks the map as non-exhaustive, indicating to the reader that there are some other
    /// elements that are not shown in the debug representation.
```

File: library/core/src/fmt/builders.rs (hunk 1 above)
Comment (tgross35) [reply library/core/src/fmt/builders.rs:796]: Ooh yes sorry I totally forgot to update the docs here

**Pull request merged**